spack/var/spack/repos/builtin/packages/py-pyprecice/package.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyPyprecice(PythonPackage):
+    """
+    This package provides python language bindings for the
+    C++ library preCICE.
+    """
+
+    homepage = "https://www.precice.org"
+    git = "https://github.com/precice/python-bindings.git"
+    url = "https://github.com/precice/python-bindings/archive/v2.0.0.1.tar.gz"
+    maintainers = ["ajaust", "BenjaminRueth"]
+
+    # Always prefer final version of release candidate
+    version("develop", branch="{{ branch }}")
+    version("2.1.1.1", sha256="972f574549344b6155a8dd415b6d82512e00fa154ca25ae7e36b68d4d2ed2cf4")
+    version("2.1.0.1", sha256="ac5cb7412c6b96b08a04fa86ea38e52d91ea739a3bd1c209baa93a8275e4e01a")
+    version("2.0.2.1", sha256="c6fca26332316de041f559aecbf23122a85d6348baa5d3252be4ddcd5e94c09a")
+    version("2.0.1.1", sha256="2791e7c7e2b04bc918f09f3dfca2d3371e6f8cbb7e57c82bd674703f4fa00be7")
+    version("2.0.0.2", sha256="5f055d809d65ec2e81f4d001812a250f50418de59990b47d6bcb12b88da5f5d7")
+    version("2.0.0.1", sha256="96eafdf421ec61ad6fcf0ab1d3cf210831a815272984c470b2aea57d4d0c9e0e")
+
+    # Import module as a test
+    import_modules = ["precice"]
+
+    patch("deactivate-version-check-via-pip.patch", when="@:2.1.1.1")
+
+    variant("mpi", default=True, description="Enables MPI support")
+
+    depends_on("mpi", when="+mpi")
+    depends_on("precice", when="@develop")
+    depends_on("precice@2.1.1", when="@2.1.1.1:2.1.1.99")
+    depends_on("precice@2.1.0", when="@2.1.0.1:2.1.0.99")
+    depends_on("precice@2.0.2", when="@2.0.2.1:2.0.2.99")
+    depends_on("precice@2.0.1", when="@2.0.1.1:2.0.1.99")
+    depends_on("precice@2.0.0", when="@2.0.0.1:2.0.0.99")
+
+    depends_on("python@3:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    depends_on("py-wheel", type="build")
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-mpi4py", type=("build", "run"), when="+mpi")
+    depends_on("py-cython@0.29:", type=("build"))
+
+    phases = ['install_lib', 'build_ext', 'install']
+
+    def build_ext_args(self, spec, prefix):
+        return [
+            "--include-dirs=" + spec["precice"].headers.directories[0],
+            "--library-dirs=" + spec["precice"].libs.directories[0]
+        ]
+
+    def install(self, spec, prefix):
+        if self.version <= Version("2.1.1.1"):
+            self.setup_py("install", "--prefix={0}".format(prefix))
```

Comment:
Does this work nownewer versions of the adapter or does it skip the installation step completely then?

Reply:
The current test runs on `develop > 2.1.1.1` and works. I ran an additional tests and to me it looks like `install` does what it is supposed to do:

Running `python3 -c "import precice; print(precice.__version__)"` not from project root, but from an independent folder works. So the package is found correctly (through the python installation), which means that the installation was executed correctly.

If I run `python3 -c "import precice; print(precice)"`, I also get a reasonable path: `<module 'precice' from '/opt/spack/var/spack/environments/precice/.spack-env/view/lib/python3.8/site-packages/precice/__init__.py'>`

I think this looks good from my perspective. We should still explicitly ask for a double check on this part, when we submit the new `package.py` to Spack, since our reviewer will probably know the answer. I myself are not 100% sure.

Reply:
That sounds good!